### DISCUSSION TITLE
po-page-dynamic-table não carrega os itens

### DISCUSSION THREAD

**Discussion by leandrodefaria** [po-ui/po-angular#1146] [Tire suas dúvidas]
Pessoal,

Estou usando o po-page-dynamic-table na qual estou consumido uma api (Não estou usando metadados), porém a lista não carrega. Se usar a api de exemplo no portal do po ui (https://po-sample-api.herokuapp.com/v1/people) retorna normalmente, mas a api que tenho não.

Vendo  na aba network do browser, percebi que o retorno da api de exemplo é dessa forma:
`
    "items": [
        {
            "id": "0148093543698",
            "name": "Ronaldo Nascimento Filho",
            "birthdate": "1980-11-01T00:00:00-00:00",
            "genre": "male",
           }]
`

A api que utilizo retorna dessa forma:
`[
    {
        "id": 3,
        "name": "teste",
        "corporate_name": null,
        "cgc": "2525569000158",
        "phone1": "31258887878"}]
`

Acredito que o problema esteja no retorno do Json  devido ao agrupamento pelo 'items'. Preciso saber se o erro é por isso e como faço para tratar essa situação, se existe alguma forma de manipular o json antes de carregar para o dynamic table

**Comment by wsteixeira** [comment]
Olá,

Até onde eu sei você não conseguira realizar esta tratativa no "po-page-dynamic-table", mas caso esteja usando o json-server para simular um servidor de API, por fazer conforme abaixo:

```
router.render = (req, res) => {
  if (res.locals.data.constructor === Array) {
    res.locals.data = { items: res.locals.data };
  }
  res.jsonp(res.locals.data);
}

```
Caso seja um servidor real, você terá que tratar direto no servidor, ou ainda usar componente "Page List" combinado com o "Table", onde poderá tratar o json no componente de serviço do angular

**Reply by leandrodefaria** [reply]
Mais fácil tratar a api.

Obrigado
